org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/JavaSearchScope.java

```diff
@@ -413,6 +415,34 @@ private int indexOf(String containerPath, String relativePath) {
 	return -1;
 }
 
+/**
+ * If the given path is internal but represents an external folder,
+ * converts it to the corresponding external path.
+ * The reference path indicates if conversion is necessary.
+ * No conversion takes place if the given path does not represent an external folder,
+ * or if conversion is unnecessary because the reference path also represents an external folder.
+ * @param given the given path to convert if necessary
+ * @param reference the reference path that indicates if conversion is necessary
+ * @return the external path that corresponds to the given path,
+ * or the given path itself if no conversion is necessary
+ */
+private String convertInternalToExternalPath(String given, String reference) {
+	IPath givenPath = new Path(given);
+	if (ExternalFoldersManager.isInternalPathForExternalFolder(givenPath)) {
+		IPath referencePath = new Path(reference);
+		if (!ExternalFoldersManager.isInternalPathForExternalFolder(referencePath)) {
+			IResource targetResource = JavaModel.getWorkspaceTarget(givenPath);
+			if (targetResource != null) {
+				IPath targetLocation = targetResource.getLocation();
+				if (targetLocation != null)
+					return targetLocation.toString();
```

Comment:
I believe we should use `toOSString()` here as that is what used in this class elsewhere for container paths